### ISSUE TITLE
removeModalAndResetState errors when close is called multiple times.

### ISSUE THREAD

**Issue by infoeon** [limonte/sweetalert2#487]
If animations are enabled and close is called multiple times or from within the resolve or reject handlers then an error is thrown. Either isVisible should check if it's closing and waiting for the animation to end, or an isClosing method should be added, or close() should check if it's already closing and be a noop.

A use case is when an alert triggers a view change and the handler wants to ensure that any open sweetalerts are closed before continuing to the new view.

Problem code:
```
function someHandler(){
    if(swal.isVisible()) swal.close();
}

swal({html: "Hello"}).then(function(){someHandler()});
```

**Comment by limonte** [comment]
Your code sample is weird :) You're trying to close the modal (`swal.close()`) right after closing it by clicking the Confirm button.

Could you please come up with the real-life code example on jsfiddle? (jsfiddle template: https://jsfiddle.net/ad3quksn/68/)

**Comment by infoeon** [comment]
Yes it's weird but the example is just a simplified version to demonstrate the issue--the use case I describe above is still very pertinent. In a single page application, if the code within the resolve handler causes a view change and the view handler has code to ensure that any open sweetalerts are closed before swapping out the UI in the DOM then the error will manifest itself. I could have a more elaborate version that demonstrates the use case but just describing what the issue and use case is should be sufficient to resolve it. In short, the currently open sweetalert should know whether it's in the process of closing and not close itself twice, or it should check to see if it's even in the DOM anymore before trying to remove itself.

**Comment by limonte** [comment]
I'll take a look on this, thanks a lot @infoeon for your feedback!

**Comment by limonte** [comment]
Hello @infoeon!

This issue is fixed in [v6.6.1](https://github.com/limonte/sweetalert2/releases/tag/v6.6.1)
